modules/katamari/src/main/ts/ephox/katamari/api/LazyValues.ts

```diff
@@ -1,7 +1,20 @@
 import { LazyValue } from './LazyValue';
 import * as AsyncValues from '../async/AsyncValues';
+import { clearTimeout, setTimeout } from '@ephox/dom-globals';
 
 /** par :: [LazyValue a] -> LazyValue [a] */
 export const par = function <T> (lazyValues: LazyValue<T>[]) {
   return AsyncValues.par(lazyValues, LazyValue.nu);
-};
\ No newline at end of file
+};
+
+export const withTimeout: <T>(baseFn: (completer: (value: T) => void) => void, ifTimeout: () => T, timeout: number) => LazyValue<T> = <T> (baseFn: (completer: (value: T) => void) => void, ifTimeout: () => T, timeout: number): LazyValue<T> =>
```

Comment:
Why put the type signature in twice? Is there any advantage to writing

```ts
export const withTimeout: [type...] = [definition with types...]
```

instead of

```ts
export const withTimeout = [definition with types...]
```

It at least looks as though the function would end up with the same type signature either way

Reply:
Good catch, we probably don't want that, as it'll likely cause confusion and also adds some maintenance overhead.

- [ ] Remove duplicate types

Reply:
Oops. No, that wasn't supposed to be there.